extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/HttpUpgradePolicy.java

```diff
@@ -0,0 +1,57 @@
+package io.quarkus.websockets.next;
+
+import io.quarkus.security.identity.SecurityIdentity;
+import io.smallrye.mutiny.Uni;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * A policy that controls which requests are allowed to upgrade the HTTP connection to a WebSocket connection.
+ * CDI beans implementing this interface are invoked on every request.
```

Comment:
We should mention that certain scopes are forbidden... at least that's what I see in build steps.